app/controllers/cockpit_controller.rb

```diff
@@ -5,15 +5,11 @@ def host_ssh_params
     render :json => SSHExecutionProvider.ssh_params(@host)
   end
 
-  def cockpit_url_for_host(host)
-    tmpl = Setting[:remote_execution_cockpit_href]
-    tmpl && tmpl != "" ? tmpl % { :host => host } : nil
-  end
-
   def redirect
-    return head(:forbidden) unless params[:redirect_uri]
+    return invalid_request unless params[:redirect_uri]
```

Comment:
changed the error code, as forbidden would more indicate permission issue, while it seems we are validating the params here.